app/routes/task_routes.py

```diff
@@ -0,0 +1,90 @@
+from app import db
+from flask import Blueprint, jsonify, make_response, request
+from app.models.task import Task
+import datetime
+from datetime import datetime as d1
+from .helpers import get_task_from_id, send_completion_slack_message
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# Task CRUD routes
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.to_dict() for task in tasks]
+    return make_response(jsonify(tasks_response), 200)
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or \
+        "completed_at" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    if request_body['completed_at']:
```

Comment:
Great work verifying valid input for `complete_at`!

Reply:
I see this same code below. Consider putting it into a helper function to DRY up your code.